cli/helper.go

```diff
@@ -32,6 +33,13 @@ func ShowHelp(cctx *ufcli.Context, err error) error {
 }
 
 func RunApp(app *ufcli.App) {
+	defer func() {
+		if r := recover(); r != nil {
+			// Generate report in LOTUS_PATH and re-raise panic
+			build.GeneratePanicReport(os.Getenv("LOTUS_PATH"), "app_panic")
```

Comment:
This isn't guaranteed to be set, and each binary may have it's own location eg `$LOTUS_MINER_PATH`, `$LOTUS_WORKER_PATH`, etc.

Have you tried to see if you can run this recovery in the `After` method of urfave cli? This will give you access to the urfave cli context which will allow you to use the cli flags and avoid using `os.Getenv`.

Being able to customize the location where the reports are generated would also be amazing. Something like `LOTUS_PANIC_REPORTS` or similar that is accepted on every binary would be great. It could default into the `LOTUS_*_PATH`. You can use `app.Name` to do a lookup for the correct cli flag to read from to get the correct repo path for defaults + overrides using env vars.

The second argument could then also be the application name as well. This in conjunction with a `LOTUS_PANIC_REPORTS` would mean someone could have all their panics send to a single location and then identified by the label.

Reply:
I used the local context instead of a lookup based on `app.Name`. LMK if you think this is sufficient. (Was trying to avoid brittle lookup regressions if names change.) I think I got all of your suggestions in here now.